lib/auth/join_ec2_test.go

```diff
@@ -169,20 +169,23 @@ func TestAuth_RegisterUsingToken_EC2(t *testing.T) {
 
 	testCases := []struct {
 		desc        string
-		tokenSpec   types.ProvisionTokenSpecV2
+		tokenSpec   types.ProvisionTokenSpecV3
```

Comment:
Since we still want to support V2 can you add some tests that also include old V2 tokens? I don't think we need anything super complex, but would be good to cover at least basic cases.